stl/inc/compare

```diff
@@ -340,9 +341,503 @@ struct compare_three_way {
     using is_transparent = int;
 };
 // clang-format on
-#endif // __cpp_lib_concepts
 
-// Other components not yet implemented
+// CONCEPT _Implicitly_convertible_to
+template <class _From, class _To>
+concept _Implicitly_convertible_to = is_convertible_v<_From, _To>;
+
+// STRUCT TEMPLATE _Choice_t
+template <class _Ty>
+struct _Choice_t {
+    _Ty _Strategy  = _Ty{};
+    bool _No_throw = false;
+};
+
+// Note: The following CPOs are passing arguments as lvalues; LWG issue not yet submitted as of 2020-10-09.
+
+// CUSTOMIZATION POINT OBJECT strong_order
+namespace _Strong_order {
+    void strong_order(); // Block unqualified name lookup; LWG issue not yet submitted as of 2020-10-09.
+
+    template <class _Ty1, class _Ty2>
+    concept _Has_ADL = requires(_Ty1&& _Left, _Ty2&& _Right) {
+        static_cast<strong_ordering>(strong_order(_Left, _Right));

```

Comment:
I am wondering why this is using a cast versus an explicit constraint on the return type
```cpp
        { strong_order(_Left, _Right) } -> strong_ordering;

```

Applies below

Reply:
Argh, forgot to remove this one after reading the standard

Reply:
There is a related question though - the Standardese asks whether `strong_­ordering(strong_­order(E, F))` is well-formed. I converted this to a `static_cast` because I dislike functional-style casts; however, the difference may be observable.

Reply:
> the difference may be observable.

Is this true when the target type is a non-reference class type that we control? My imagination is lacking today.

Reply:
I will resolve this as we can't imagine a way to observe any difference from the Standardese. (For the record, as @miscco realized, explicitly constraining the return type would prevent the function call from returning something not `strong_ordering` but convertible to `strong_ordering`; the functional-style cast and `static_cast` permit that.)